data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/AwsCloudMapPeerListProviderTest.java

```diff
@@ -65,6 +67,7 @@ void setUp() {
         awsServiceDiscovery = mock(ServiceDiscoveryAsyncClient.class);
         namespaceName = RandomStringUtils.randomAlphabetic(10);
         serviceName = RandomStringUtils.randomAlphabetic(10);
+        queryParameters = Collections.emptyMap();
```

Comment:
This is using as input the default value for the query parameters. So the tests below are not verifying that that the `queryParameters` is being used. You should generate a more random map.

Relatedly, we should have some tests which explicitly test when the value is `null`.

Reply:
Good point, I will generate a random map.

`constructor_throws_with_null_QueryParameters()` is explicitly testing for this value being null. Or are you referring to adding a test for when `pluginSetting.getTypedMap` returns null?